amethyst_core/src/transform/systems.rs

```diff
@@ -38,25 +33,14 @@ impl<'a> System<'a> for TransformSystem {
     type SystemData = (
         Entities<'a>,
         ReadExpect<'a, ParentHierarchy>,
-        ReadStorage<'a, Transform>,
+        WriteStorage<'a, Transform3<f32>>,
```

Comment:
If systems are just going to hardcode the generic, what's the utility of having it generic at all? I see many potential problems of different systems trying to use either f32 of f64, which will end up being a totally separate components.

Reply:
We would need to propagate the generic with bounds to every single system that uses transforms (which will probably be a lot). It will add a lot of noise and possible source of bugs, yet I'm not sure what are the upsides to counter that.

Reply:
that's me being dumb, I forgot to update it correctly.
What I'm trying to go for is forcing all systems to have this bound:

```
N: Real = f32
```
That way you don't need additional code if you want the default f32, and you can still enable f64 easily when creating the systems.

Reply:
You will have to remember to register the right component system types to the ECS though. It's not really obvious to newcommers that all of those types must match, otherwise things will silently fail to see each other.

Reply:
There isn't. That's the point. You might also actually have both at the same time if you so desire. I'm just in general not sure what's the point of having support for `f64` at all. I might be not up to date. Were there any usecases presented for that?

I fear that this will push people into unnecessarily implementing their own systems in a generic way. Just because hardcoding the type might feel like a tech debt, people will often do it. But in reality it might be totally unnecessary to care about generics here at all. This problem might probably be solved by good documentation. It just feels like a bunch of unnecessary noise to me anyway.

Reply:
Also note that this might have an unitentional implact on other teams down the road. I suspect that this `Transform` component will be used to position things in the editor for example. If you want to be oblivious to the chosen type, you have to somehow support both. What if an entity has both at the same time? Also any other components that end up being often combined with Transform might end up needing the same generic in order to effectively do math between them without precision problems. It is just very "infectious".

Another example is the system I'm currently working on - encoders. I would have to implement the encoder in generic way and explain that the registration needs the correct Real type in order to take any effect. Now supporting `Transform<f64>` actually requires changes to the shader source code - all my matrices must use double precision. This quickly gets out of hands.

Reply:
I haven't read through everything here yet, just wanted to chime in on the aspect of supporting `f64`. Star Citizen's team went as far as to say 
> 64-bit world space is one of the driving elements behind making Star Citizen possible.
[Source](https://www.gamersnexus.net/gg/2622-star-citizen-sean-tracy-64bit-engine-tech-edge-blending)

Reply:
The original issue in this thread has also been fixed.